packages/ts/form/src/Models.ts

```diff
@@ -27,29 +23,32 @@ export function hasFromString<T>(model: AbstractModel<T>): model is AbstractMode
 
 export type Value<M> = M extends AbstractModel<infer T> ? T : never;
 
+export interface HasValue<T> {
+  value?: T;
+}
+
+export type ModelParent<T> = AbstractModel | HasValue<T>;
+
+export type ModelConstructor<M> = new (parent: { value: undefined }, key: 'value', optional: boolean) => M;
```

Comment:
It looks like an `EmptyModelConstructor`, not just a `ModelConstructor`. What about changing the name?

Reply:
Also, I consider `value` is a misleading name. I spent quite a time to understand that it is about the empty parent, not the real model. Let's have more special name here, like `$value$` I introduced.

Reply:
Let's address the `value` name confusion later.